hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/crypto/OpensslCipher.c

```diff
@@ -169,9 +169,19 @@ JNIEXPORT void JNICALL Java_org_apache_hadoop_crypto_OpensslCipher_initIDs
                       "EVP_CIPHER_CTX_set_padding");
   LOAD_DYNAMIC_SYMBOL(dlsym_EVP_CIPHER_CTX_test_flags, env, openssl,  \
                       "EVP_CIPHER_CTX_test_flags");
+// name changed in OpenSSL 3 ABI - see History section in EVP_EncryptInit(3)
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+  LOAD_DYNAMIC_SYMBOL(dlsym_EVP_CIPHER_CTX_block_size, env, openssl,  \
```

Comment:
let's have a single block for defining the string and then use 
```
#if  OPENSSL_VERSION_NUMBER >= 0x30000000L
#define CTX_BLOCK_SIZE "EVP_CIPHER_CTX_get_block_size"
#define CTX_ENCRYPTING "EVP_CIPHER_CTX_is_encrypting"
...
#else
#define CTX_BLOCK_SIZE "EVP_CIPHER_CTX_block_size"
#define CTX_ENCRYPTING "EVP_CIPHER_CTX_encrypting"
...
#endif

 LOAD_DYNAMIC_SYMBOL(dlsym_EVP_CIPHER_CTX_block_size, env, openssl,
                      CTX_BLOCK_SIZE);
```

Then the actual uses just refer to the macro definition as needed

1. makes things less complex
2. lines up for openssl 4 and any more changes.

Reply:
Will have a look this weekend.

Reply:
The requested change would not work on OpenSSL 1.0: `EVP_CIPHER_CTX_encrypting` is only available on OpenSSL 1.1. Would we need to support OpenSSL 1.0 still?

Reply:
Wait- I think I misunderstood you. We can decouple of course. Pushed a change addressing this and rebased as well.